src/buildApplication/generateFunctionsMap.ts

```diff
@@ -203,71 +205,66 @@ function extractWebpackChunks(
 } {
 	const webpackChunks = new Map<number, string>();
 
-	const parsedContents = parse(functionContents, {
-		ecmaVersion: 'latest',
-		sourceType: 'module',
-	}) as Node & { body: LooseNode[] };
-
-	const expressions = parsedContents.body
-		.filter(
-			({ type, expression }) =>
-				type === 'ExpressionStatement' &&
-				expression?.type === 'CallExpression' &&
-				expression.callee?.type === 'MemberExpression' &&
-				expression.callee.object?.type === 'AssignmentExpression' &&
-				expression.callee.object.left?.object?.name === 'self' &&
-				expression.callee.object.left.property?.name === 'webpackChunk_N_E' &&
-				expression.arguments?.[0]?.elements?.[1]?.type === 'ObjectExpression'
-		)
-		.map(
-			node => node?.expression?.arguments?.[0]?.elements?.[1]?.properties
-		) as LooseNode[][];
-
-	for (const objectOfChunks of expressions) {
-		for (const chunkExpression of objectOfChunks) {
-			const key = chunkExpression?.key?.value;
-			if (key in existingWebpackChunks) {
-				if (
-					existingWebpackChunks.get(key) !== generate(chunkExpression.value)
-				) {
+	const parsedContents = recast.parse(functionContents, { parser: acornParser })
+		.program as AstTypes.ProgramKind;
+
+	const chunksProperties = parsedContents.body.map(
+		(statement: AstTypes.ExpressionStatementKind) => {
+			if (
+				statement.type === 'ExpressionStatement' &&
+				statement.expression?.type === 'CallExpression' &&
+				statement.expression.callee?.type === 'MemberExpression' &&
+				statement.expression.callee.object.type === 'AssignmentExpression' &&
+				statement.expression.callee.object.left.type === 'MemberExpression' &&
+				statement.expression.callee.object.left.object.type === 'Identifier' &&
+				statement.expression.callee.object.left.object.name === 'self' &&
+				statement.expression.callee.object.left.property.type ===
+					'Identifier' &&
+				statement.expression.callee.object.left.property.name ===
```

Comment:
If the several checks above feel cumbersome, it might be possible to leverage the `visit` function of `recast` to `visitIdentifer`, check if it's name is 'webpackChunk_N_E', and then operate on the resulting node. Not sure if that is selective enough to work with this use case (like perhaps you really only need instances where it's the lh of an expression), but figured I'd suggest it :D

Reply:
yeah it actually feels very cumbersome, I didn't touch it as I am planning to use the parsing in the hacks we've got here: https://github.com/cloudflare/next-on-pages/blob/119b55eac1e592d07546541d50c00dde2f293ba4/src/buildApplication/generateFunctionsMap.ts#L238

so I was thinking of creating utilities that make this sort of thing less cumbersome there (so that I can create the utils and make sure they work nicely in both places)